### ISSUE TITLE
tfe_project - invalid value for name for simple project name with "-" in it

### ISSUE THREAD

**Issue by hajdukda** [hashicorp/terraform-provider-tfe#748]
#### Terraform Cloud/Enterprise version
Terraform Cloud

#### Terraform version
1.3.6

#### Terraform Configuration Files
```
resource "tfe_project" "projects" {
  for_each = toset(local.projects)

  name         = each.value
  organization = local.tfe_organization
}
```

#### Debug Output
N/A

#### Expected Behavior
Project is created with specified name

#### Actual Behavior
Creation throws an error "Error creating the new project Internal-Services EU: invalid value for name"

#### Additional Context
There is no information about forbidden characters in documentation: https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/project

**Comment by brandonc** [comment]
Hey, thanks for reporting this. This was fixed in [this upstream issue](https://github.com/hashicorp/go-tfe/issues/607) but the bump hasn't made it to a terraform-provider-tfe release yet.

**Comment by brandonc** [comment]
This was fixed in release [v0.41.0](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs)
